index.js

```diff
@@ -71,9 +77,38 @@ class App extends MatrixPuppetBridgeBase {
   }
   sendReadReceiptAsPuppetToThirdPartyRoomWithId() {
     // no-op for now
+  }
+  
+  sendImageMessageAsPuppetToThirdPartyRoomWithId(id, data) {
+    return download.getTempfile(data.url, { tagFilename: true }).then(({path}) => {
+      const img = path;
+      let fs = require('fs');
+      let image = fs.readFileSync(img);
+      return this.client.sendMessage( id, data.text, [{contentType : data.mimetype, size : data.size, data : image} ] );
+    });  
   }
+
+stringToArrayBuffer(str) {
+    if (typeof str !== 'string') {
+        throw new Error('Passed non-string to stringToArrayBuffer');
+    }
+    var res = new ArrayBuffer(str.length);
+    var uint = new Uint8Array(res);
+    for (var i = 0; i < str.length; i++) {
+        uint[i] = str.charCodeAt(i);
+    }
+    return res;
+}
+
   sendMessageAsPuppetToThirdPartyRoomWithId(id, text) {
-    return this.client.sendMessage(id, text);
+console.log(this.client.syncGroups());
+console.log(id);
+    if ( id.match(/^÷/) ) {
+      console.log("sending to group");
+      return this.client.sendMessageToGroup(this.stringToArrayBuffer(id), text);
```

Comment:
You shouldn't need to define a function to convert strings to ArrayBuffers if TextEncoder is available in our JavaScript execution environment. Try removing `stringToArrayBuffer` and using it, something like this:
```suggestion
      let idAsArrayBuffer = new TextEncoder().encode(id).buffer;
      return this.client.sendMessageToGroup(idAsArrayBuffer, text);
```